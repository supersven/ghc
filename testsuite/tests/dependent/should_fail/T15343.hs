{-# LANGUAGE RankNTypes #-}
{-# LANGUAGE ScopedTypeVariables #-}
{-# LANGUAGE TypeApplications #-}
{-# LANGUAGE TypeInType #-}
module T15343 where

import Data.Kind

elimSing :: forall (p :: forall z. z). p
elimSing = undefined

data WhySym :: Type -> Type

hsym = elimSing @WhySym
