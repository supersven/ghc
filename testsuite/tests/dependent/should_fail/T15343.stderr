
T15343.hs:14:18: error:
    • Expecting one more argument to ‘WhySym’
      Expected kind ‘forall z. z’, but ‘WhySym’ has kind ‘* -> *’
    • In the type ‘WhySym’
      In the expression: elimSing @WhySym
      In an equation for ‘hsym’: hsym = elimSing @WhySym
